File: app/src/main/java/com/legue/axel/lolsummonertool/database/dao/match/ParticipantDao.java
package com.legue.axel.lolsummonertool.database.dao.match;

import androidx.lifecycle.LiveData;
import androidx.room.Dao;
import androidx.room.Insert;
import androidx.room.OnConflictStrategy;
import androidx.room.Query;
import androidx.room.Update;


import com.legue.axel.lolsummonertool.database.model.match.Participant;

import java.util.List;

@Dao
public interface ParticipantDao {

    @Query("SELECT * FROM participants ORDER BY id")
    LiveData<List<Participant>> getParticipants();

    @Query("SELECT * FROM participants WHERE matchId = :matchId")
    LiveData<Participant> getParticipant(long matchId);

    @Insert(onConflict = OnConflictStrategy.REPLACE)
    void insertAllParticipants(List<Participant> participants);

    @Query("DELETE FROM participants")
    void deleteAll();

    @Update(onConflict = OnConflictStrategy.REPLACE)
    void updateParticipant(Participant participant);
}

File: app/src/main/java/com/legue/axel/lolsummonertool/database/dao/match/ParticipantDao.kt
package com.legue.axel.lolsummonertool.database.dao.match


import androidx.lifecycle.LiveData
import androidx.room.*
import com.legue.axel.lolsummonertool.database.model.match.Participant

@Dao
interface ParticipantDao {

    @Query("SELECT * FROM participants ORDER BY id")
    fun getParticipants(): LiveData<List<Participant>>

    @Query("SELECT * FROM participants WHERE matchId = :matchId")
    fun getParticipant(matchId: Long): LiveData<Participant>

    @Insert(onConflict = OnConflictStrategy.REPLACE)
    fun insertAllParticipants(participants: List<Participant>)

    @Query("DELETE FROM participants")
    fun deleteAll()

    @Update(onConflict = OnConflictStrategy.REPLACE)
    fun updateParticipant(participant: Participant)
}
